src/traces/scatter3d/attributes.js

```diff
@@ -101,8 +104,10 @@ module.exports = {
         width: scatterLineAttrs.width,
         dash: scatterLineAttrs.dash
     },
-    marker: {  // Parity with scatter.js?
-        color: scatterMarkerAttrs.color,
+    // Currently, `marker` etc. contain mixed attributes (colorAttributes
+    // and a bunch of others); maybe a future API change could wrap parts
+    // that belong together, under a `colorspec` or similar attribute name.
```

Comment:
Now the API is 'flat': e.g. a JSON would say:
```
"marker": {
   "symbol": whatever,
   "colorscale": "Greens",
   "cauto": true
}
```
i.e. the properties relating to color and properties intrinsic to the marker are mixed. A possible (though incompatible) alternative is to keep them organized according to these style groups, e.g. 
```
"marker": {
   "symbol": whatever,
   "colorspec": {
     "colorscale": "Greens",
     "cauto": true
  }
}
```
(btw. I'm planning to remove the code comment itself)